Automattic/jetpack
# Add product cards in recommendation flow

#### Changes proposed in this Pull Request:

This PR adds product cards in the recommendations flow that suggest products based  what products/ plan the site already has. These product cards show in the right-hand side of the recommendations cards.

#### Jetpack product discussion


pbNhbs-1iw-p2#comment-7724

#### Does this pull request change what data or activity we track or use?


Yes, this PR adds one new analytics event:

The `jetpack_recommendations_step_product_card_button_click ` is a new event that is recorded when the new `<ProductSpotlight \>` card component CTA is clicked. It records the recommendation step that the CTA button was clicked on.

#### Testing instructions:






* Set up your test site with this PR. Test first with a site that has a Jetpack free plan.
* Navigate to `/wp-admin/admin.php?page=jetpack#/recommendations/publicize`
* You should see the product card for Jetpack Security showing on the right-hand side of the step, it should look like this:

![Screen Shot 2022-04-22 at 8 51 29 AM](https://user-images.githubusercontent.com/18016357/164717993-72fb6e3a-5478-437c-9982-efa3d71e707c.png)

* Now, check that the product card for Jetpack Security is also showing also on `/wp-admin/admin.php?page=jetpack#/recommendations/security-plan`. The copy for the card should be different on this screen:

![Screen Shot 2022-04-22 at 8 51 17 AM](https://user-images.githubusercontent.com/18016357/164718059-0bc5349a-3690-44d5-a41f-6889a63bf417.png)

* On both recommendation steps, the CTA button for "Get Jetpack Security" should link to a checkout page with the `Jetpack Security (10GB)` plan in the cart for the site you are testing with. These links should open in a new tab.
* Confirm that the new `jetpack_recommendations_step_product_card_button_click` analytics event is recorded after clicking the CTA buttons on the cards. The step the CTA was clicked on should be passed as a prop to the event.
* Now, repeat this test with a test site that has a Security or Complete plan. The product card should NOT show on either of the recommendation steps.
